lightning/src/ln/onion_utils.rs

```diff
@@ -2814,6 +2814,24 @@ fn process_failure_packet(
 	update_attribution_data(onion_error, shared_secret, hold_time);
 }
 
+pub(crate) fn process_fulfill_attribution_data(
+	attribution_data: Option<&AttributionData>, shared_secret: &[u8], hold_time: u32,
+) -> AttributionData {
+	let mut attribution_data =
+		attribution_data.map_or(AttributionData::new(), |attribution_data| {
+			let mut attribution_data = attribution_data.clone();
+
+			attribution_data.shift_right();
+
+			attribution_data
+		});
+
+	attribution_data.update(&[], &shared_secret, hold_time);
```

Comment:
Can you comment for why we always pass in an empty slice here?

Reply:
Added